WordPress/Classes/ViewRelated/Reader/Detail/ReaderDetailViewController.swift

```diff
@@ -813,6 +815,31 @@ class ReaderDetailViewController: UIViewController, ReaderDetailView {
         static let delay: Double = 50
         static let preferredToolbarHeight: CGFloat = 58.0
     }
+
+    // MARK: - Managed object observer
+
+    func startObservingPost() {
```

Comment:
Since we're using `SwiftUI`, could we achieve the same by using `Post` as `ObservedObject`? I only find one place where it's done the way I imagine - `SiteDomainsView`.

Reply:
We could observe the object directly, but that causes a couple of issues:

- The crash you just fixed from observing a managed object directly would reappear here. We would need to check if the object is faulted before rendering and if the object was deleted, the entire header would disappear for the user.
- The header isn't entirely written in `SwiftUI` so when `configure` is called, there are callbacks to `UIKit` which invalidate the containing view's size and lays out the view again. We could save that callback in the `SwiftUI` view and then call it when needed, but I don't see a good way to determine when that's needed. It would require some hacky code to determine if the view's height changed, and then we could call it. In my opinion, that doesn't seem worth dealing with. Ideally, we would get to the point where all these views were written in `SwiftUI`.

Reply:
Got it. Thank you for explanation!